src/main/java/core/basesyntax/strategy/impl/SupplyOperationHandler.java

```
@@ -0,0 +1,26 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+
+public class SupplyOperationHandler implements OperationHandler {
+    private final StorageDao storageDao;
+
+    public SupplyOperationHandler(StorageDao storageDao) {
+        if (storageDao == null) {
+            throw new RuntimeException("Cannot access to storage dao: parameter is null");
+        }
+        this.storageDao = storageDao;
+    }
```

Although the constructor checks `storageDao` for null (good), the handler should not rely on the underlying DAO to re-validate inputs. Keep handler-level validation focused and fail fast so invalid transactions never reach the DAO. This aligns with single responsibility and clear error locations in stack traces.